### PR TITLE
#16817 now when a contentlet is locked by admin, limited user can not…

### PR DESCRIPTION
**Pull request by jdotcms**
… destroy a contentlet

### CODE REVIEW

**Comment by jdotcms** [comment]
When a contentlet is locked by an admin and a limited user tries to destroy the contentlet, it must be not allowed.
Now when a limited user tries to destroy a locked contentlet, throws an error

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 1]
```diff
@@ -2010,7 +2010,13 @@ private boolean internalDestroy (final List<Contentlet> contentlets, final User
                 throw new DotContentletStateException(CAN_T_CHANGE_STATE_OF_CHECKED_OUT_CONTENT);
             }
 
-            this.canLock(contentlet, user);
+            final boolean bringOldVersions  = false;  // we do not want old version in order to be more efficient
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: I think this boolean declaration is not needed. The comment is enough

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 2]
```diff
@@ -2010,7 +2010,13 @@ private boolean internalDestroy (final List<Contentlet> contentlets, final User
...
+            final List<Contentlet> versions =  this.findAllVersions(APILocator.getIdentifierAPI().find(contentlet.getIdentifier()), bringOldVersions,
+                    user, respectFrontendRoles);
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: This logic requires IT

Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: I agree also about an integration test

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: I just want to be clearer on my intention, it is quite the same in terms of memory and so

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java (hunk 2 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: Done

File: dotCMS/src/integration-test/java/com/dotcms/datagen/LanguageCodeDataGen.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    @Override
+    public String next() {
+
+        final int    languageIndex = this.random.nextInt(languageCodes.length);
+        final String languageCode  = this.languageCodes[languageIndex];
+        return languageCode;
+    }
+
+    @Override
+    public String persist(String object) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/LanguageCodeDataGen.java]: Codacy Issue found: [Parameter 'object' is not assigned and could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4632905)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/LanguageCodeDataGen.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    final String[] languageCodes = Locale.getISOLanguages();
+    final Random   random        = new Random();
+
+    @Override
+    public String next() {
+
+        final int    languageIndex = this.random.nextInt(languageCodes.length);
+        final String languageCode  = this.languageCodes[languageIndex];
+        return languageCode;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/LanguageCodeDataGen.java]: Codacy Issue found: [Consider simply returning the value vs storing it in local variable 'languageCode'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4632905)

**Review by erickgonzalez — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**